homeassistant/components/ollama/conversation.py

```diff
@@ -151,7 +152,8 @@ async def async_process(
                 # Make a copy of the messages because we mutate the list later
                 messages=list(message_history.messages),
                 stream=False,
-                keep_alive=KEEP_ALIVE_FOREVER,
+                # keep_alive requires specifying unit. In this case, seconds
+                keep_alive=f"{settings.get(CONF_KEEP_ALIVE, DEFAULT_KEEP_ALIVE)}s",
```

Comment:
I would prefer that if we want to switch to the default of Ollama, we should just not pass in the option at all and also not offer it to the user.

@synesthesiam since you picked this current default, thoughts?

Reply:
Understandable that there are many, many knobs that could be turned on Ollama and its probably better to not overwhelm the user.

My thinking on this one was that different users could have pretty substantially different setups and usage patterns. I did a decent amount of testing, and at least on my hardware (1080ti, i7-4770K, SSD), the latency from loading the llama3 model into memory was surprisingly negligible. However, having 44% of my GPU RAM tied up by something only used a handful of times per day was much more limiting. I'd can imagine at least 3 classes of users:

1) High performance seekers: potentially serving many users and likely using dedicated hardware. Current value of -1 probably best fit.
2) Hobbyists/recent adopters: probably deploying on a desktop computer they use for daily work/computing. Having a big chunk of GPU RAM tied up probably infeasible if they ever want to play a video game or other intensive tasks. Values of 0 - X minutes probably ideal.
3) Budget homelabbers with either a raspberry pi or mini pc. Potentially deploying Ollama using CPU instead of GPU. I'm actually not sure what the performance characteristics would look like here. I'd _guess_ values of -1 or 0 would be undesirable here but not sure.

I also wanted to leave room for customization as new models come out. Even if llama3:8B loads up quick, it's possible models in the next few months get unwieldy. I can envision someone running a massive model on a device with unified memory wanting to avoid reading from disk.

Reply:
So I am leaning to just letting users use `OLLAMA_KEEP_ALIVE`. Mike is currently out so let's wait until he is back next Thursday.

Reply:
I set the keep alive to -1 with the assumption that for HA users would not want to pay the cost of loading a model every time it wasn't used for 5 minutes. When used in a voice pipeline, the delay of a few seconds is a bit annoying.
I'm fine making this configurable, but I'd like to keep the default at -1 even if Ollama themselves use 5 minutes.

Reply:
But we could let users configure Ollama with -1, instead of having this option in HA

Reply:
We could, but I would argue Ollama's default is not right for HA. I was confused when first building the integration because my models kept getting unloaded.

Reply:
Ultimately deferring to you both, my two cents is that Ollama can be used by multiple users for multiple services. The right default behavior for a Home Assistant only setup won't necessarily scale to other configurations. Having Home Assistant impose an immutable default behavior that impacts the whole system running Ollama might be a bit overkill. 

The other edge case is that setting -1 in the Home Assistant request doesn't actually guarantee the model will remain indefinitely in memory. If the Ollama host receives a Home Assistant request is followed by another request (e.g., Open Webui), it will overwrite the -1 with with the value passed by the other service. Which can put you in a situation where the defaults of each service ping pong and overwrite each other. In Open Webui, keep_alive is [configurable](https://github.com/open-webui/open-webui/blob/9e4dd4b86f77653b59dc1fa97fe3f72e8252b359/src/lib/components/chat/Chat.svelte#L634)

Perhaps being user-configurable like Open Webui is the more congruent approach? I'd be happy to modify this PR to simply switch the default from 5m to -1

Reply:
I'm fine with making the keep alive configurable, but I think we should default to -1 in HA

Reply:
I've pushed two commits to this PR:
1. Pull in the latest dev branch 
2. Update default value to -1

I've also updated the documentation PR to reflect this update in the same way: https://github.com/home-assistant/home-assistant.io/pull/33191

Please let me know if you need anything further.